include/lsst/afw/geom/SpherePoint.h

```diff
@@ -0,0 +1,343 @@
+         */
+        Angle getLongitude() const noexcept { return _longitude*radians; };
+
+        /**
+         * @brief The latitude of this point.
+         *
+         * @return the latitude, in the interval [-&pi;/2, &pi;/2] radians.
+         *
+         * @par Exception Safety
+         * This method shall not throw exceptions.
+         */
+        Angle getLatitude() const noexcept { return _latitude*radians; };
+
+        /**
+         * @brief A unit vector representation of this point.
+         *
+         * @return a unit vector whose direction corresponds to this point
+         *
+         * @par Exception Safety
+         * This method shall not throw exceptions.
+         */
+        Point3D getVector() const noexcept;
+
+        /**
+         * @brief A representation of this point as a <tt>[longitude, latitude]</tt> array.
+         *
+         * @param index the index of the spherical coordinate to return. Must
+         *              be either 0 or 1.
+         *
+         * @return @ref getLongitude() if @c index = 0, or @ref getLatitude()
+         *         if @c index = 1
+         *
+         * @throws OutOfRangeError Thrown if @c index is neither 0 nor 1.
+         *
+         * @par Exception Safety
+         * The program state shall be unchanged in the event of an exception.
+         */
+        Angle operator[](size_t index) const;
+
+        /**
+         * @brief Tests whether this point is either coordinate pole.
+         *
+         * @return @c true if this point is at the north or south pole,
+         *         @c false otherwise
+         *
+         * @par Exception Safety
+         * This method shall not throw exceptions.
+         */
+        bool atPole() const noexcept;
+
+
+        /**
+         * @brief Tests whether this point is a well-defined position.
+         *
+         * @return @c true if @ref getLongitude(), @ref getLatitude(), and
+         *         @ref getVector() return finite floating-point values;
+         *         @c false if any return NaN or infinity.
+         *
+         * @par Exception Safety
+         * This method shall not throw exceptions.
+         */
+        bool isFinite() const noexcept;
+
+        /*
+         * Comparisons between points
+         */
+
+        /**
+         * @brief Test whether two points represent the same position.
+         *
+         * To make equality transitive (as well as consistent with built-in
+         * types), two objects are considered equal if and only if their
+         * floating-point fields are exactly equal.
+         *
+         * @param other the point to test for equality
+         * @return true if this point matches @c other exactly, false otherwise
+         *
+         * @par Exception Safety
+         * This operator shall not throw exceptions.
+         */
+        bool operator==(SpherePoint const & other) const noexcept;
+
+        /**
+         * @brief Test whether two points represent the same position.
```

Comment:
I agree it's not optimal, but I still think it is better not to duplicate information. Also the doc for `operator==` should be right before the doc for `operator!=` so it will be easy to find (in fact the user will likely have already seen it)